examples/vision/forwardforward.py

```diff
@@ -0,0 +1,352 @@
+"""
+Title: Using Forward-Forward Algorithm for Image Classification
+Author: [Suvaditya Mukherjee](https://twitter.com/halcyonrayes)
+Date created: 2022/12/21
+Last modified: 2022/12/23
+Description: Training a Dense-layer based model using the Forward-Forward algorithm.
+
+"""
+
+"""
+## Introduction
+
+The following example explores how to use the Forward-Forward algorithm to perform
+training instead of the traditionally-used method of backpropagation, as proposed by
+[Prof. Geoffrey Hinton](https://www.cs.toronto.edu/~hinton/FFA13.pdf)
+The concept was inspired by the understanding behind [Boltzmann
+Machines](http://www.cs.toronto.edu/~fritz/absps/dbm.pdf). Backpropagation involves
```

Comment:
Please keep all markdown links on a single line, otherwise they won't properly render

Reply:
Will make this change across the file